packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,278 @@
+   */
+  public readonly cluster: Cluster;
+  /**
+   * IAM role of the instance profile for the nodegroup
+   */
+  public readonly role: IRole;
+
+  private readonly desiredSize: number;
+  private readonly maxSize: number;
+  private readonly minSize: number;
+
+  constructor(scope: Construct, id: string, props: NodegroupProps ) {
+    super(scope, id);
+
+    this.cluster = props.cluster;
+
+    this.desiredSize = props.desiredSize ?? props.minSize ?? 2;
+    this.maxSize = props.maxSize ?? this.desiredSize;
+    this.minSize = props.minSize ?? 1;
+
+    if (this.desiredSize > this.maxSize) {
+      throw new Error(`Desired capacity ${this.desiredSize} can't be greater than max size ${this.maxSize}`);
+    }
+    if (this.desiredSize < this.minSize) {
+      throw new Error(`Minimum capacity ${this.minSize} can't be greater than desired size ${this.desiredSize}`);
+    }
+
+    const ngRole = new Role(this, 'NodeGroupRole', {
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com')
+    });
+
+    ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'));
+    ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'));
+    ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'));
+
+    const resource = new CfnNodegroup(this, id, {
+      clusterName: this.cluster.clusterName,
+      nodegroupName: props.nodegroupName,
+      nodeRole: ngRole.roleArn,
+      subnets: this.cluster.vpc.selectSubnets(props.subnets).subnetIds,
+      amiType: props.amiType,
+      diskSize: props.diskSize,
+      forceUpdateEnabled: props.forceUpdateEnabled ?? true,
+      instanceTypes: props.instanceType ? [props.instanceType.toString()] : undefined,
+      labels: props.labels,
+      releaseVersion: props.releaseVersion,
+      remoteAccess: props.remoteAccess ? {
+        ec2SshKey: props.remoteAccess.ec2SshKey,
+        sourceSecurityGroups: props.remoteAccess.sourceSecurityGroups ?
+          props.remoteAccess.sourceSecurityGroups.map(m => m.securityGroupId) : undefined
+      } : undefined,
+      scalingConfig: {
+        desiredSize: this.desiredSize,
+        maxSize: this.maxSize,
+        minSize: this.minSize
+      },
+      tags: props.tags
+    });
+
+    // do not attempt to map the role if `kubectl` is not enabled for this
+    // cluster or if `mapRole` is set to false. By default this should happen.
+    if (this.cluster.kubectlEnabled) {
+      // see https://docs.aws.amazon.com/en_us/eks/latest/userguide/add-user-role.html
+      this.cluster.awsAuth.addRoleMapping(ngRole, {
+        username: 'system:node:{{EC2PrivateDNSName}}',
+        groups: [
+          'system:bootstrappers',
+          'system:nodes'
+        ]
+      });
+    } else {
+      // since we are not mapping the instance role to RBAC, synthesize an
+      // output so it can be pasted into `aws-auth-cm.yaml`
+      new CfnOutput(this, 'InstanceRoleARN', {
```

Comment:
Allow users to opt out of this (see eks.Cluster)

Reply:
Looks like the managed ng will take care of the aws-auth-cm for us and we don't have to deal with it in CDK. I am working on it to get things sorted.

Reply:
OK. I can confirm the managed NG will auto-map the RBAC under the hood and we don't need to deal with it anymore. So by default we don't have to print the `InstanceRoleARN` in the output now.